spring-boot/src/main/java/com/azure/samples/todo/DefaultToDoService.java

```diff
@@ -1,17 +1,19 @@
 package com.azure.samples.todo;
 
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import jakarta.transaction.Transactional; 
+import jakarta.inject.Inject;
 
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 @Service
+@Transactional
```

Comment:
I don’t think Spring recognizes this. You’ll need to use the Spring annotation.

Reply:
Yes, we have to add @EnableTransactionManagement to the application to enable it. 
jakarta.transaction.Transactional can be recognized by Spring, see https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html

Reply:
Spring Boot will detect spring-jdbc on the classpath and postgresql and will create a DataSource and a JdbcTemplate automatically. See https://spring.io/guides/gs/managing-transactions
Spring can recognize jakarta.transaction.Transactional, see https://docs.spring.io/spring-framework/reference/data-access/transaction/declarative/annotations.html

Reply:
Here list some debug log:
```
2024-12-31T15:30:48.752+08:00  INFO 25449 --- [Todo] [nio-8080-exec-8] c.azure.samples.todo.DefaultToDoService  : Getting all items
2024-12-31T15:30:48.788+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-8] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll]
Hibernate: select tdi1_0.id,tdi1_0.completed,tdi1_0.description from to_do_item tdi1_0
2024-12-31T15:30:48.960+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-8] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll]
2024-12-31T15:30:55.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Getting transaction for [com.azure.samples.todo.DefaultToDoService.addToDoItem]
2024-12-31T15:30:55.448+08:00  INFO 25449 --- [Todo] [nio-8080-exec-2] c.azure.samples.todo.DefaultToDoService  : Adding item: ToDoItem[id=null, desciption=REST APITRE, completed=false ]
2024-12-31T15:30:55.449+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
Hibernate: select nextval('to_do_item_seq')
2024-12-31T15:30:55.475+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
2024-12-31T15:30:55.475+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-2] o.s.t.i.TransactionInterceptor           : Completing transaction for [com.azure.samples.todo.DefaultToDoService.addToDoItem]
Hibernate: insert into to_do_item (completed,description,id) values (?,?,?)
2024-12-31T15:30:59.423+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Getting transaction for [com.azure.samples.todo.DefaultToDoService.updateToDoItem]
2024-12-31T15:30:59.423+08:00  INFO 25449 --- [Todo] [nio-8080-exec-4] c.azure.samples.todo.DefaultToDoService  : Updating item: ToDoItem[id=1, desciption=, completed=false ]
2024-12-31T15:30:59.424+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Getting transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
Hibernate: select tdi1_0.id,tdi1_0.completed,tdi1_0.description from to_do_item tdi1_0 where tdi1_0.id=?
2024-12-31T15:30:59.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Completing transaction for [org.springframework.data.jpa.repository.support.SimpleJpaRepository.save]
2024-12-31T15:30:59.445+08:00 TRACE 25449 --- [Todo] [nio-8080-exec-4] o.s.t.i.TransactionInterceptor           : Completing transaction for [com.azure.samples.todo.DefaultToDoService.updateToDoItem]
```